pylint/checkers/variables.py

```diff
@@ -671,17 +672,21 @@ def _inferred_to_define_name_raise_or_return(
                 for handler in handlers
             )
 
-        if isinstance(node, (nodes.With, nodes.For, nodes.While)):
-            return NamesConsumer._defines_name_raises_or_returns_recursive(name, node)
-
-        if isinstance(node, nodes.Match):
-            return all(
-                NamesConsumer._defines_name_raises_or_returns_recursive(name, case)
-                for case in node.cases
-            )
+        match node:
+            case nodes.With() | nodes.For() | nodes.While():
+                return NamesConsumer._defines_name_raises_or_returns_recursive(
+                    name, node
+                )
 
-        if not isinstance(node, nodes.If):
-            return False
+            case nodes.Match():
+                return all(
+                    NamesConsumer._defines_name_raises_or_returns_recursive(name, case)
+                    for case in node.cases
+                )
+            case nodes.If():
+                pass
+            case _:  # pragma: no cover
+                return False
```

Comment:
This can't be tested either. The function is only called for `Try`, `With`, `For`, `While`, `Match` and `If` nodes. All of which are being handled above.

_Didn't remove it just to make it clear in the function that there are no other node types._

Reply:
```suggestion
            case _:  # pragma: no cover
                # The function is only called for Try, With, For, While, Match and 
                # If nodes. All of which are being handled above.
                return False
```

Reply:
Should I add an `assert False` as well instead of `return False`?

Reply:
Make sense !